projects/plugins/jetpack/extensions/blocks/cookie-consent/edit.js

```diff
@@ -29,10 +139,86 @@ function CookieConsentBlockEdit( { attributes, setAttributes } ) {
 		} );
 	}
 
+	async function goToTemplatePart( savedTemplate ) {
+		setIsSaving( false );
+		removeBlock( clientId );
+		window.open(
+			`/wp-admin/site-editor.php?postType=${ savedTemplate.type }&postId=${ savedTemplate.id }`
+		);
+	}
+
 	const blockProps = useBlockProps( {
 		className: `wp-block-jetpack-cookie-consent align${ align }`,
 	} );
 
+	const isInWarningState = parentPost?.slug !== PART_SLUG;
+
+	/* If the block is added in the right place (in its own part), mark it as such, this is needed in the save function */
+	useEffect( () => {
+		if ( parentPost ) {
+			setAttributes( { isInWarningState } );
+		}
+	}, [ setAttributes, parentPost, isInWarningState ] );
+
+	/* If the block is added in the wrong place (not in its own part), render UI that helps the user create a template part. */
+	if ( isInWarningState ) {
+		if ( mode === 'LOADING' ) {
+			return __( 'Loading…', 'jetpack' );
+		} else if ( mode === 'PART_EXISTS' ) {
+			return (
+				<Card>
+					<CardBody>
+						<p>
+							{ __(
+								'In order to be rendered on every page of your site, the Cookie Consent Block is best added as a template part. You already have created a template part before.',
+								'jetpack'
+							) }
+						</p>
+					</CardBody>
+					<CardFooter>
+						<Button
+							variant="primary"
+							disabled={ isSaving }
+							onClick={ () => {
+								setIsSaving( true );
+								goToTemplatePart( part );
+							} }
+						>
+							{ __( 'Go to template part', 'jetpack' ) }
+						</Button>
+					</CardFooter>
+				</Card>
+			);
+		} else if ( mode === 'PART_DOES_NOT_EXIST' ) {
+			return (
+				<Card>
+					<CardBody>
+						<p>
+							{ __(
+								'In order to be rendered on every page of your site, the Cookie Consent Block is best added as a template part.',
```

Comment:
```suggestion
								'It looks like you are trying to add a cookie consent banner to your page. This needs to be put into it's own template part in order for it to work. Click the button below to do that for you!',
```

Maybe someone else can weigh on this as well. I think this should be a bit more direct so they don't just leave the cookie block shell and save the page thinking that will work.

Reply:
> It looks like you are trying to add a cookie consent banner to your page

The issue is that I don't want people to feel they made a mistake. Because there is no other way to add a cookie consent. I want to convey that it's part of the plan to add the block, then create a part. Maybe something like:

![image](https://github.com/Automattic/jetpack/assets/17054134/1526fe62-0654-47a4-8c47-1058f1ec8b8b)

Reply:
I see what you mean. I think the terms that I thought would be problematic were "is best added" or "should". These imply that its an option to where they want it to go which is false. Your first option is probably good if you changed "is best added as" to "needs to be".

```suggestion
								'In order to be rendered on every page of your site, the Cookie Consent Block needs to be a template part.',
```